### ISSUE TITLE
Ads don't work for me

### ISSUE THREAD

**Issue by BossCode45** [monkeytypegame/monkeytype#3351]
### Did you clear cache before opening an issue?

- [X] I have cleared my cache

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does the issue happen when logged in?

Yes

### Does the issue happen when logged out?

Yes

### Does the issue happen in incognito mode when logged in?

Yes

### Does the issue happen in incognito mode when logged out?

Yes

### Account name

boss45

### Account config

{"theme":"comfy","themeLight":"serika","themeDark":"serika_dark","autoSwitchTheme":false,"customTheme":false,"customThemeColors":["#323437","#e2b714","#e2b714","#646669","#2c2e31","#d1d0c5","#ca4754","#7e2a33","#ca4754","#7e2a33"],"favThemes":[],"showKeyTips":true,"showLiveWpm":true,"showTimerProgress":true,"smoothCaret":true,"quickRestart":"off","punctuation":false,"numbers":false,"words":25,"time":30,"mode":"words","quoteLength":[3],"language":"english","fontSize":"15","freedomMode":false,"difficulty":"normal","blindMode":false,"quickEnd":false,"caretStyle":"default","paceCaretStyle":"default","flipTestColors":false,"layout":"default","funbox":"none","confidenceMode":"off","indicateTypos":"off","timerStyle":"mini","colorfulMode":false,"randomTheme":"off","timerColor":"main","timerOpacity":"1","stopOnError":"off","showAllLines":false,"keymapMode":"off","keymapStyle":"staggered","keymapLegendStyle":"lowercase","keymapLayout":"overrideSync","fontFamily":"Roboto_Mono","smoothLineScroll":false,"alwaysShowDecimalPlaces":false,"alwaysShowWordsHistory":false,"singleListCommandLine":"manual","capsLockWarning":true,"playSoundOnError":false,"playSoundOnClick":"off","soundVolume":"0.5","startGraphsAtZero":true,"showOutOfFocusWarning":true,"paceCaret":"off","paceCaretCustomSpeed":100,"repeatedPace":true,"pageWidth":"125","chartAccuracy":true,"chartStyle":"line","minWpm":"off","minWpmCustomSpeed":100,"highlightMode":"letter","alwaysShowCPM":false,"ads":"sellout","hideExtraLetters":false,"strictSpace":false,"minAcc":"off","minAccCustom":90,"showLiveAcc":false,"showLiveBurst":false,"monkey":true,"repeatQuotes":"off","oppositeShiftMode":"off","customBackground":"","customBackgroundSize":"cover","customBackgroundFilter":[0,1,1,1,1],"customLayoutfluid":"qwerty#dvorak#colemak","monkeyPowerLevel":"off","minBurst":"off","minBurstCustomSpeed":100,"burstHeatmap":false,"britishEnglish":false,"lazyMode":false,"showAverage":"off","tapeMode":"off"}

### Current Behavior

When I enable ads it just shows a shaded box with the label ad.
When attempting to view video ads it just shows a loading circle in a shaded box but nothing happens.
Currently using sellout mode but normal mode is the same.

### Expected Behavior

Ads should display so I can support the developer for free!!!

### Steps To Reproduce

1. Enable adds
2. Finish a test so that ads should appear
3. Try to watch a video add

### Environment

- OS: Linux (Arch on LTS kernel)
- Browser: Firefox
- Browser Version: 102.0 (64-bit) Mozilla Firefox for Arch Linux (latest from arch repos)

### Anything else?

Not super urgent but I'd like to have ads as I don't have enough money to donate but love the site.

**Comment by Miodec** [comment]
Make sure you are not using any extensions that block cookie consent popups. Consent from the user is needed for the ads to appear.

**Comment by BossCode45** [comment]
Yes i whitelisted it on my ad blocker and also the bug occurs in private mode (where I have no ad blockers).

**Comment by Miodec** [comment]
Private mode firefox has an aggressive anti-tracker feature which also gets rid of the consent popup.

**Comment by karimk123** [comment]
I'm also having the same problem but sometimes they show up and sometimes they don't, I don't have any type of blocker.
![image](https://user-images.githubusercontent.com/64856262/181005935-a75e8976-35ea-4789-ab7f-9bd426e4d139.png)

**Comment by Miodec** [comment]
> I'm also having the same problem but sometimes they show up and sometimes they don't, I don't have any type of blocker. ![image](https://user-images.githubusercontent.com/64856262/181005935-a75e8976-35ea-4789-ab7f-9bd426e4d139.png) And I already got the cookie consent pop up and I accepted, so that isn't the problem.

I think the ads are still slowly being ramped up, so sometimes it might take a while for them to load and sometimes so placements wont be filled at all.

**Comment by karimk123** [comment]
One day later, now they're not loading at all and they haven't loaded once the whole day.

**Comment by krishnasantosh11** [comment]
Same Problem.

Is it possible to load the ads before hand? Instead of ads loading on the results page, load them on the typing page itself and they can be served on the results page.

![Screenshot_20220728_213013](https://user-images.githubusercontent.com/75202541/181585576-211fa16d-0b11-42f9-8b41-8fd8af3956fa.png)

**Comment by Miodec** [comment]
> One day later, now they're not loading at all and they haven't loaded once the whole day.

Can you show me the whole console output? Are there any other errors besides the webconfig one?

**Comment by karimk123** [comment]
Another day later now only one of the 2 ads loads and its the same exact ad every time so that kind of defeats the purpose.
But here is the full console:
![image](https://user-images.githubusercontent.com/64856262/181638113-f1f05809-e036-418d-ba13-42dd0e1e99af.png)

**Comment by Miodec** [comment]
@karimk123 where are you from? maybe the location has something to do with it

**Comment by BossCode45** [comment]
![image](https://user-images.githubusercontent.com/74084650/182045435-81ffedbb-4931-4ad3-aa16-60e08f0afc5d.png)
I get this error in the console (I'm guessing when it tries to update ads). Video ads also aren't popping up for me. I am from NZ if that helps.

**Comment by Miodec** [comment]
Is it any better now guys? August was early days for ads so is there any improvement?

**Comment by karimk123** [comment]
For me it works most of the time but sometimes its just one ad out of the 2 on the page, and sometimes none of them load at all.

**Comment by hama73** [comment]
To confirm this issue, I have had the ads on for a week now, and I have visited the site almost every day at different times, and I have only seen ads three times, with only one or two of the banners showing. I tried all the different modes, Also for the video ad all I see is a loading circle. 

Also, the ones I saw did not instantly load after completing a test on the result page. I was keeping the tab open, and after about an hour or so, when I was returning, one of them was showing, and it would've stayed there until I refreshed the site. I see Google ads elsewhere almost when I load the sites,
 
![ads_don't_work_monkeytype](https://user-images.githubusercontent.com/96790752/216233885-0e274cea-b124-4428-bdec-19e7ec8a166a.png)

 - [x] Cleared cache
 - [x] Issue occurred when logged in and out, as well as in Incognito mode.
 - [x] Accepted all the cookies, even the ones labelled under "Advertising."

 OS: Linux (Kubuntu 22.10)
 Browser: Chrome Version 109.0.5414.119 (Official Build) (64-bit)

**Comment by Miodec** [comment]
@hama73 where are you from?

**Comment by hama73** [comment]
I'm now in Iraq, and I just connected to US VPN, and all of the ads loaded normally, including the video ad, just as you said.

> maybe the location has something to do with it

**Comment by Miodec** [comment]
Okay, ill see what i can do

**Comment by karimk123** [comment]
I've just tried to use a VPN to the US but i'm not even getting any ads. Weird

**Comment by beh-10257** [comment]
![image](https://user-images.githubusercontent.com/77834479/222963501-d6f4fa69-4462-4e92-ac45-175807640673.png)
well until this screenshot was taken there were no ads

![image](https://user-images.githubusercontent.com/77834479/222963532-e40a2bca-5817-45b7-b714-37d298ab8f56.png)
after this an added pop up
well I'm using Firefox and disabling enhanced tracking protection in monkeytype
+ using a vpn to usa
+ video ads aren't working at all
+ I am initially from north africa well I guess there is a problem from the cookies
+ I didn't log in while doing this test
+ the ad just hides after 40 seconds

**Comment by Miodec** [comment]
Hey everyone. Im currently testing another ad provider. Its randomly selected on page load, you can see the choice at the top of the console 
![image](https://user-images.githubusercontent.com/13181393/223398480-e8d02dfa-bbb4-4ba9-a41b-03840d007203.png)

Options are eg (old) and pw (new)

Do you see any improvements?

**Comment by beh-10257** [comment]
so now It works perfectly with the new ad provider
I'm from north Africa so yeah
no need to use any vpn just disable ublock origin and no need to disable enhanced protection in firefox

if you are using clearurls extention it will diable the ads regardless of the ad provider or your preferences
+ by my experience the cookies will not affect if the ads will show up or not

**Comment by Miodec** [comment]
@beh-10257 Good to hear. By the way, you can force the new ad provider to be always used by pasting this into the console `window.localStorage.setItem("adChoiceForce","pw")`

**Comment by beh-10257** [comment]
@Miodec yeah It works now thanks but for some reson not like yesterday enhanced protection should be disabled

I think this situation could be improved (I'm not a developer sorry if that's too much) but this is better than nothing thanks

**Comment by hama73** [comment]
In my case, it was showing this message yesterday,

![image](https://user-images.githubusercontent.com/96790752/223927411-f60acffd-12ff-4115-8261-0146449367fb.png)

I didn't have any kind of blockers, So I cleared the cache, accepted all cookies, enabled "Intrusive Ads," "Pop-ups and Redirects" in the site settings, thinking they might be related, and even forced the new Ad provider, but nothing changed, On Chrome and Edge, I experienced the same behavior. 

But Today with the new provider "pw" everything loads as expected, apart from the video ad. 
In the beginning, the banners would be filled in 2 to 5 seconds. after that they would change regularly.

**Comment by Miodec** [comment]
Yeah i fixed a small issue around midnight. It should all be good now with PW.

**Comment by beh-10257** [comment]
yeah it works nicely thanks
